File: QE-Java Selenium/Activity7.py
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.support.ui import WebDriverWait
from selenium.webdriver.support import expected_conditions as EC

driver = webdriver.Firefox()
wait = WebDriverWait(driver, 10)

driver.get("https://training-support.net/webelements/dynamic-controls")

print("Title:", driver.title)

input_field = driver.find_element(By.XPATH, '//*[@id="textInput"]')

print("Input field enabled (before):", input_field.is_enabled())

driver.find_element(By.XPATH, '//*[@id="textInputButton"]').click()

wait.until(
    EC.element_to_be_clickable((By.XPATH, '//*[@id="textInput"]'))
)

print("Input field enabled (after):", input_field.is_enabled())

driver.quit()

File: QE-Java Selenium/Activity7.java
package demos;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;

import java.time.Duration;

public class Activity7 {
    public static void main(String[] args) {

        WebDriver driver = new FirefoxDriver();
        WebDriverWait wait = new WebDriverWait(driver, Duration.ofSeconds(10));

        driver.get("https://training-support.net/webelements/dynamic-controls");

        System.out.println("Title: " + driver.getTitle());

        System.out.println("Input field enabled (before): " +driver.findElement(By.xpath("//*[@id='textInput']")).isEnabled());

        driver.findElement(By.xpath("//*[@id='textInputButton']")).click();

        wait.until(ExpectedConditions.elementToBeClickable(By.xpath("//*[@id='textInput']")));

        System.out.println("Input field enabled (after): " +driver.findElement(By.xpath("//*[@id='textInput']")).isEnabled());

        driver.quit();
    }
}
